itests/framework/framework.go

```diff
@@ -128,7 +129,7 @@ func FullNodeAndMiner(t *testing.T) (*kit.TestFullNode, *kit.TestMiner) {
 	fnOpts := []kit.NodeOpt{
 		kit.ConstructorOpts(
 			lnode.Override(new(lp2p.RawHost), func() (host.Host, error) {
-				return libp2p.New()
+				return libp2p.New(libp2p.DefaultTransports)
```

Comment:
This is a hack to deal with the following issue:
- You cannot start a listener on the same address multiple times (e.g. starting two quic-v1 listeners on `/ip4/1.2.3.4/udp/1234` will [panic](https://github.com/libp2p/go-libp2p/blob/686a32aae741aeb0ebe825553ec066d669a744ae/p2p/transport/quic/virtuallistener.go#L60))
- `libp2p.New()` listens on all interfaces by default
- There's a lotus hook that starts listening https://github.com/filecoin-project/lotus/blob/9e4f1a4d23ad72ab191754d4f432e4dc754fce1b/node/modules/lp2p/addrs.go#L92

We're getting around this by passing the same default transports we would have but no addresses which is making sure the libp2p host constructor doesn't do any listening, only the lotus code that explicitly listens does the listening. Setting any transports or listen addresses would do here though.

Note: The code in lotus that calls the listener was removed in kubo here https://github.com/ipfs/kubo/pull/9857#discussion_r1186109527 (and I tagged the lotus go-libp2p PR). Probably that code can be removed as well given the same caveats as the kubo removal.

This should be fine for this PR, but it seems like probably this should get cleaned up. Maybe just porting the fix from kubo to lotus is sufficient, but perhaps changes in other modules (e.g. go-libp2p, boost, or the lotus DI setup used for testing) would also be helpful.

Heads up @marten-seemann @marcopolo @magik6k